### ISSUE TITLE
.POWER STATUS

### ISSUE THREAD

**Issue by robbiet480** [robbiet480/CEC-Roomie#3]

**Comment by robbiet480** [comment]
I just contacted Roomie to request they add Power and Input feedback support.

For the power feedback, I requested that they parse a human readable string and provided some suggestions:
- "on" and "off"
- "poweron" and "poweroff"
- "on" and "standby"

But I left the final decision to them, as cec-web can output any string that I want.

**Comment by robbiet480** [comment]
Roomie got back to me and said their policy is to not implement custom feedback for custom devices. I asked them if there are any feedback formats already implemented that I could piggyback on top of (i.e. I can make cec-web say anything in the HTTP response, so is there a format I can use that is from another vendor that is easy enough to duplicate?) Waiting to hear back.

**Comment by robbiet480** [comment]
Still waiting for Roomie, but I went ahead and put `.POWER STATUS`, `.INPUT STATUS` and `.VOLUME STATUS` back into the main plist

**Comment by noelhibbard** [comment]
Have you had any luck with feedback in Roomie? I built a custom device with a command for .POWER STATUS which queries a URL and when I check my Apache log I can see where Roomie is in fact hitting the server but it isn't displaying the status in the remote. I've modified my PHP code to return on/off, true/false, 0/1, yes/no and nothing seems to make Roomie happy. It's a bit ridiculous that Roomie support doesn't address this. I know they are all about plug in play but when it comes to home automation, we aren't the typical crowd that needs plug in play. They need to get with it and make this work. Sorry to hijack your thread. I was just curious if you've had any luck.

**Comment by robbiet480** [comment]
@noelhibbard Yeah I heard back a while ago and forgot to update this. There is no support in Roomie (when I spoke to them about 6 months ago) for custom devices supporting any STATUS commands (INPUT, POWER, MUTE etc). Will let you know when/if I hear more!

**Comment by noelhibbard** [comment]
Jesus this company pisses me off. Here we are freaking years later and I still haven't been able to implement two way communication. Have you been able to figure anything out? I wrote a REST server for controlling everything in my home so I can respond with anything I want but Simple Control seems to refuse to document this crap. I really don't understand their reasoning in restricting their product. Hmm.. Simple Control does support two way communication on Pioneer AVRs I believe. Pioneer has great documentation on their products. I bet I could spoof a Pioneer AVR and then connect Simple Control to that.

**Comment by noelhibbard** [comment]
Geez that was easy. I spoofed a Pioneer AVR and Simple Control liked it. The protocol is very easy to duplicate. Here is a link to Pioneers documentation:
https://www.pioneerelectronics.com/StaticFiles/PUSA/Files/Home%20Custom%20Install/FY16AVR_External_Command_for_CI_2.xlsx

I didn't bother with implementing autodiscovery so I had to add the device manually in Simple Control. My test server is running on TCP:60128 so in Simple Control I manually added a device by IP and typed the poty 60128. Then I selected Receiver and then Pioneer then  "All Models Zone 1" and then it started sending queries to my server. So Simple Control sends a ?P\r and I respond with POW0\r\n (yeah 0 = On), then it sends a ?V\r and I respond with VOL160\r\n and it displayed -70.5dB as the volume level in Simple Control.
